### PR TITLE
[WIP] Added test cases for pureauth

### PR DESCRIPTION
**Pull request by krausese**
Added the test case pureauth, 

Fixes https://github.com/koppor/jabref/issues/74

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/test/java/org/jabref/logic/util/BracketedPatternTest.java [hunk 1]
```diff
@@ -59,13 +59,30 @@ public void nullDatabaseExpansionTest() {
         BracketedPattern pattern = new BracketedPattern("[year]_[auth]_[firstpage]");
         assertEquals("2017_Kitsune_213", pattern.expand(bibentry, another_database));
     }
+    
+    @Test
+    public void pureauthTest() {
+        BibDatabase another_database = new BibDatabase();
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/util/BracketedPatternTest.java]: Please don't use underscores, but camel case, i.e., `anotherDatabase`. (here and in the test below)

Comment (krausese) [reply src/test/java/org/jabref/logic/util/BracketedPatternTest.java]: Thanks for your reply. I will change it.

File: src/test/java/org/jabref/logic/util/BracketedPatternTest.java [hunk 2]
```diff
@@ -59,13 +59,30 @@ public void nullDatabaseExpansionTest() {
...
+        assertEquals("", pattern.expand(bibentry, anotherDatabase));
+    }
 
     @Test
     public void emptyDatabaseExpansionTest() {
         BibDatabase another_database = new BibDatabase();
         BracketedPattern pattern = new BracketedPattern("[year]_[auth]_[firstpage]");
         assertEquals("2017_Kitsune_213", pattern.expand(bibentry, another_database));
     }
+
```
Comment (LinusDietz) [inline src/test/java/org/jabref/logic/util/BracketedPatternTest.java]: can you remove these two new newlines?

**Comment by LinusDietz** [comment]
Side note: https://github.com/koppor/jabref/issues/74 is only partially fulfilled by this pull request. Before closing that issue there should also be tests for [pureauthors3]. But that doesn't affect this PR.

However, @krausese  if you want to do a follow-up PR, feel free to go ahead (;

File: src/test/java/org/jabref/logic/util/BracketedPatternTest.java [hunk 3]
```diff
@@ -189,7 +204,7 @@ public void testResolvedParentNotInDatabase() {
         database.removeEntry(dbentry);
         database.insertEntry(child);
 
-        assertEquals("", BracketedPattern.expandBrackets("[author]", ';', child, database));
-    }
+		assertEquals("", BracketedPattern.expandBrackets("[author]", ';', child, database));
```
Comment (LinusDietz) [inline src/test/java/org/jabref/logic/util/BracketedPatternTest.java]: now the indentation is wrong

**Comment by koppor** [comment]
This one should be carefully reviewed if pureauth really brings the correct results. Only from the diff, I cannot do it. This PR is a good start and it should continued to work on.

**Comment by koppor** [comment]
The test missed a non-set author, but a set editor. I added such a test, which showed that `pureauth` did not work. I fixed it at https://github.com/JabRef/jabref/pull/3903.

I close this PR as https://github.com/JabRef/jabref/pull/3903 is the follow up.

**Pull request closed without merging**